hypothesis-python/src/hypothesis/database.py

```diff
@@ -278,6 +380,44 @@ def delete(self, key: bytes, value: bytes) -> None:
         except OSError:
             pass
 
+    def _start_listening(self) -> None:
+        try:
+            from watchdog.events import (
+                FileCreatedEvent,
+                FileDeletedEvent,
+                FileSystemEvent,
+                FileSystemEventHandler,
+            )
+            from watchdog.observers import Observer
+        except ImportError:
+            warnings.warn(
+                f"listening for changes in a {self.__class__.__name__} "
+                "requires the watchdog library. To install, run "
+                "`pip install hypothesis[watchdog]`",
+                stacklevel=4,
+            )
+            return
+
+        _broadcast_changed = self._broadcast_changed
+
+        class Handler(FileSystemEventHandler):
+            def on_any_event(self, event: FileSystemEvent) -> None:
+                _broadcast_changed()
+
+        self._observer = Observer()
+        self._observer.schedule(
+            Handler(),
+            self.path,  # type: ignore # upstream type is too narrow (str only)
```

Comment:
Link (and if necessary open) an upstream issue, so we can tell when it's safe to remove the ignore?

Reply:
was easy enough to send them a pull: https://github.com/gorakhargosh/watchdog/pull/1096